Added planner unit testing framework

AbhishekMathur11 opened the pull request:
Unit testing thingy, useful hopefully `\__(*\*)_/'

Commit:
Add planner unit testing framework with yaml-cpp dependency and build configuration

AbhishekMathur11 committed:
Update CMakeLists.txt

tomputer-g commented on vader_planner/CMakeLists.txt:
```diff
@@ -93,72 +57,30 @@ generate_messages(
-
-## Generate dynamic reconfigure parameters in the 'cfg' folder
+
+find_package(yaml-cpp REQUIRED)
```
Why is this included like 4 times, can you remove the duplicates

tomputer-g commented on vader_planner/CMakeLists.txt:
```diff
@@ -271,8 +193,14 @@ add_dependencies(vader_svd_single_arm_planner ${${PROJECT_NAME}_EXPORTED_TARGETS
 add_dependencies(gazebo_assassin ${${PROJECT_NAME}_EXPORTED_TARGETS})
+planner_tester
 target_link_libraries(gazebo_assassin ${catkin_LIBRARIES})
+planner_tester
```
Same here - why is this there three times

tomputer-g commented on vader_planner/config/planner_tester_ompl.yaml:
```diff
@@ -0,0 +1,60 @@
+RRTstar:
```
Are these all the parameters that RRTStar takes? Can you put a documentation link to wherever you found these (I believe yaml has # commenting)

AbhishekMathur11 replied:
Mainly this page: https://ompl.kavrakilab.org/RRTstar_8h_source.html and otherwise there are a couple of other params, but none seemed to be influencing the planner as much as these, so for now these should be good enough to make the planner robust. I'll add it in the READme as well

tomputer-g commented on vader_planner/src/planner_tester.cpp:
```diff
@@ -0,0 +1,814 @@
+    int attempts = 0;
+    int max_attempts = NUM_RANDOM_BLOCKS * 50; // Prevent infinite loops
+    
+    while(generated_blocks.size() < NUM_RANDOM_BLOCKS && attempts < max_attempts)
```
So this _attempts_ to generate (50*35) blocks. When it fails, it doesnt try to generate another one (loop continues another attempt)
1) can we make the random seed deterministic? When we find a good random blocks setup, we can keep it static ish
2) how do we know if the preset goal point is still reachable after randomly generating blocks?

tomputer-g reviewed (commented):
Looking good, left some comments, at most minor changes

AbhishekMathur11 committed:
Update planner_tester.cpp

Made block generation more deterministic and around the arm initial position with a radius, and the goal point outside the radius, and so there is higher reachability guarantee, and also more better insights on parameters will be obtained.

AbhishekMathur11 committed:
Update CMakeLists.txt

Removed the yaml-cpp repetitive includes

AbhishekMathur11 committed:
Update ReadMe.md

AbhishekMathur11 committed:
Update ReadMe.md

AbhishekMathur11 committed:
Update ReadMe.md

AbhishekMathur11 committed:
Update ReadMe.md

AbhishekMathur11 committed:
Update ReadMe.md

AbhishekMathur11 commented:
Resolved the stuff
